src/main/java/io/debezium/connector/jdbc/dialect/mysql/MySqlDatabaseDialect.java

```diff
@@ -190,4 +194,10 @@ protected void addColumnDefaultValue(SinkRecordDescriptor.FieldDescriptor field,
         }
         super.addColumnDefaultValue(field, columnSpec);
     }
+
+    @Override
+    public Set<Class<? extends Exception>> communicationExceptions() {
+        return Collect.unmodifiableSet(LockAcquisitionException.class,
```

Comment:
I had a look to the code and seems that those exceptions come from Hibernate that tries to uniform the different dialect exception. So in that case I think these can be moved to the `GeneralDatabaseDialect`.

I'll leave the last word to @Naros

Reply:
@mfvitale sure, I agree.
I'll modify test code from mysql failure case to common jdbc case to apply to `GeneralDatabaseDialect`.

Reply:
Hi guys, sorry for the late reply.  

Yes, Hibernate does unify exceptions for a variety of reasons; it makes the application code more portable across different database vendors, and secondly, it allowed the same code to be portable between JPA and non-JPA compliant applications in the past when the Hibernate ORM session didn't implement the JPA `EntityManager` contract, when that was an entirely separate module in the ORM portfolio.

So I think for `LockAcquisitionException` this is fine.  

One thing to note moving forward is to be wary about what exceptions we add here.  The JPA spec is pretty clear that in some circumstances, its generally best to discard everything, including the session under specific exception scenarios.  We use a stateless session so we may be able to weasel around that in many cases, but it's just worth being aware if we add to this list and we begin to see strange rollback/retry behavior.